doc for kubectl-gs cluster updates (#1248)
giantswarm/docs: src/content/ui-api/kubectl-gs/update-cluster.md

---
linkTitle: update cluster 
title: "'kubectl gs update cluster' command reference"

> description: Reference documentation on how to update a Cluster using 'kubectl gs'.

Reviewer: ```suggestion
description: Reference documentation on how to upgrade a workload cluster using kubectl-gs.
```

> description: Reference documentation on how to update update a Cluster using 'kubectl gs'.

Author: ```suggestion
description: Reference documentation on how to update a Cluster using 'kubectl gs'.
```

weight: 100
menu:
  main:

>     parent: uiapi-kubectlgs

Reviewer: ```suggestion
    identifier: kubectl-gs-update-cluster
    parent: uiapi-kubectlgs
```

This is required in this case to make the item appear in the menu. Without this, HUGO issues a warning

```
WARN 2021/12/14 11:11:09 "/Users/marian/git/giantswarm/docs/src/content/ui-api/kubectl-gs/update-cluster.md:1:1": duplicate menu entry with identifier "update cluster" in menu "main"
```

and does not add the page to the menu.

aliases:
  - /reference/kubectl-gs/update-cluster/
last_review_date: 2021-12-13
owner:
  - https://github.com/orgs/giantswarm/teams/team-phoenix
user_questions:
  - How can I update a cluster from the command line?
---

# `kubectl gs update cluster`

> Updating clusters can be done in two ways. A cluster can be immediately updated or a cluster update can be scheduled in the future.

Reviewer: ```suggestion
This command allows to upgrade a workload cluster. Upgrades can either be triggered immediately, or the upgrade can be scheduled to happen at a specific date and time.
```

## Usage

> ### Update a cluster immediately

Reviewer: ```suggestion
### Upgrade a cluster immediately
```

```nohighlight

> kgs update cluster --name a1b2c --namespace org --release-version 16.1.0 --provider aws

Reviewer: - We don't use the `kgs` alias any more.
- Long commands should be broken into one flag per line
- Better use a more realistic example namespace

```suggestion
kubectl gs update cluster \
  --name a1b2c \
  --namespace org-acme \
  --release-version 16.1.0 \
  --provider aws
```

```

> ### Schedule a cluster update

Reviewer: ```suggestion
### Schedule a cluster upgrade
```

```nohighlight
kgs update cluster --name a1b2c --namespace org --release-version 16.1.0 --scheduled-time "2022-01-01 02:00" --provider aws
```

Supported flags:

> - `--name` - ID of the cluster.

Reviewer: ```suggestion
- `--name` - Name of the cluster.
```

> - `--namespace` - Namespace of the cluster.

Reviewer: ```suggestion
- `--namespace` - Namespace of the cluster resource.
```

- `--release-version` - Version of the release the cluster should be upgraded to.

> - `--scheduled-time` - Optionally: Scheduled time when cluster should be updated, time format 'YYYY-MM-DD HH:MM'.

Reviewer: ```suggestion
- `--scheduled-time` (optional): Scheduled time when cluster should be updated, in the format 'YYYY-MM-DD HH:MM'. Timezone UTC is assumed. If not given, the upgrade happens as soon as possible.
```
